test/clients/openshift/template.go

```diff
@@ -31,3 +45,46 @@ func (cli *Client) InstantiateTemplate(srcTemplateName, dstNamespace string) err
 
 	return wait.PollImmediate(2*time.Second, 10*time.Minute, ready.CheckTemplateInstanceIsReady(cli.TemplateV1.TemplateInstances(dstNamespace), dstNamespace))
 }
+
+// InstantiateTemplateFromBytes instantiates an openshift template from a byte slice
+func (cli *Client) InstantiateTemplateFromBytes(yamlData []byte, dstNamespace string) error {
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	// Decode the YAML to an object.
+	var list corev1.List
+	_, _, err := s.Decode(yamlData, nil, &list)
+	if err != nil {
+		return err
+	}
+
+	// Some types, e.g. List, contain RawExtensions.  If the appropriate types
+	// are registered, these can be decoded in a second pass.
+	for i, o := range list.Items {
```

Comment:
none of this is needed - you should be able to post the template you unmarshal verbatim